### DISCUSSION TITLE
A pagebreak show rule for headings influeces the destination of heading references

### DISCUSSION THREAD

**Discussion by y2-Whyto** [typst/typst#5234] [Q&A]
Hello everyone,

I'm a newbie struggling and searching to learn typst. I'm using it to take notes.

For formatting reasons, I set a couple of show rules and set rules. Some of which I think are reletive to my issue are:

```typst
#show heading.where(level: 1): it => {
  pagebreak(weak: true)
  align(center, text[#it])
}

#show heading.where(level: 2): it => {
  pagebreak(weak: true)
  it
}

#set heading(
  numbering: (..nums) => 
  if nums.pos().len() == 1 {
      "第" + nums.pos().map(str).first() + "章"
    } else {
      nums.pos().map(str).join(".")
    },
  supplement: none
)

#show ref: it => {
  let el = it.element
  it + {
    if el.func() == heading and el.level != 1 {"节"}
  }
}
```

Then I notice that, if I try to refenrence a Level 1 or 2 heading, when I click on the link in the rendered PDF, it just jump to the page right before the heading where I want it to land. But instead, I hope it jumps to the location of the heading.

Is there any solution? Please help 🙏🏻

**Comment by laurmaedje** [comment]
This might be fixed in Typst 0.12, you could try with 0.12.0-rc2.

**Reply by y2-Whyto** [reply]
OK, thanks! I'll have a try.

**Reply by y2-Whyto** [reply]
Hi, I just tried again with 0.12.0-rc2. Unfortunately this issue remained the same – jumping to the page before the heading. Actually I don't really know if it is a bug or a *feature*? But from a user's perspective like mine, this could be more likely classified as a bug. Anyways, I haven't found any workaround for it.

**Reply by laurmaedje** [reply]
In principle, Typst should skip over the pagebreak when determining the heading's position. Could you share a full standalone example that exhibits the problem?

**Reply by y2-Whyto** [reply]
Sorry for the delay. This is a simplified example of my project, where I reference headings at the end of "ch2.typ". From my side, when I click on refs, the issue happens.

[example.zip](https://github.com/user-attachments/files/17444461/example.zip)
